tests/Unit/Http/Controllers/TestControllerTest.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Tests\Unit\Http\Controllers;
+
+use Tests\TestCase;
+use App\Http\Controllers\TestController;
+use App\Repositories\Test\TestRepositoryInterface;
+use Illuminate\Database\Eloquent\Collection;
+use Mockery as M;
+use Illuminate\Http\Request;
+use Illuminate\Http\RedirectResponse;
+use App\Models\Test;
+
+
+class TestControllerTest extends TestCase
+{
+    protected $testMock;
+    protected $testController;
+
+    public function setUp(): void
+    {
+        $this->testMock = M::mock(TestRepositoryInterface::class);
+        $this->testController = new TestController($this->testMock);
+        parent::setUp();
+    }
+
+    public function tearDown(): void
+    {
+        unset($this->testController);
+        M::close();
+        parent::tearDown();
+    }
+
+    public function testIndexFunction()
+    {
+        $this->testMock
```

Comment:
em test những case nào trong toàn bộ file này

Reply:
do trong các function của controller này chỉ có 2 cases: 1 là tiếng việt, 2 là tiếng anh. 2 cases này em dùng để hiển thị datetime theo đúng format của mỗi ngôn ngữ. Nên khi test các function của controller này, function nào có xử lý về ngôn ngữ thì function đó em sẽ chia ra thành 2 test